documentation/docs/extending/camera-manager.mdx

```diff
@@ -0,0 +1,185 @@
+   * @example
+   * ```js
+   * // store position, target
+   * const { position, target } = cameraManager.getCameraState();
+   * // restore position, target
+   * cameraManager.setCameraState({ position, target });
+   * ```
+   */
+  setCameraState(state: CameraState): void;
+
+  /**
+   * Get camera's state
+   * @returns Camera state: position, target and rotation.
+   */
+  getCameraState(): Required<CameraState>;
+
+  /**
+   * Subscribes to changes of the camera event. This is used by Reveal to react on changes of the camera.
+   * @param event Name of the event.
+   * @param callback Callback to be called when the event is fired.
+   */
+  on(event: 'cameraChange', callback: CameraChangedEvent): void;
+  /**
+   * Unsubscribes from changes of the camera event.
+   * @param event Name of the event.
+   * @param callback Callback function to be unsubscribed.
+   */
+  off(event: 'cameraChange', callback: CameraChangedEvent): void;
+
+  /**
+   * Moves camera to a place where the content of a bounding box is visible to the camera.
+   * @param box The bounding box in world space.
+   * @param duration The duration of the animation moving the camera.
+   * @param radiusFactor The ratio of the distance from camera to center of box and radius of the box.
+   */
+  fitCameraToBoundingBox(boundingBox: THREE.Box3, duration?: number, radiusFactor?: number): void;
+  /**
+   * Updates internal state of camera manager. Expected to update visual state of the camera
+   * as well as it's near and far planes if needed. Called in `requestAnimationFrame`-loop.
+   * Reveal performance affects frequency with which this method is called.
+   * @param deltaTime Delta time since last update in seconds.
+   * @param boundingBox Global bounding box of the model(s) and any custom objects added to the scene.
+   */
+  update(deltaTime: number, boundingBox: THREE.Box3): void;
+  /**
+   * @obvious
+   */
+  dispose(): void;
+}
+
+```
+
+Main implementation specific functions are `setCameraState`, `fitCameraToBoundingBox` and `update`. 
+Setting state shouldn't be possible when `rotation` and `target` are passed at the same time because they could conflict.
+When implementing these functions you can use a helper class `CameraManagerHelper` that contains some useful methods.
+
+## Example implementation
+
+Here is an example implementation of a custom camera manager that utilizes standard ThreeJS `OrbitControls` for mouse movement:
+
+```js
+import * as THREE from 'three';
+import { OrbitControls } from 'three/examples/jsm/controls/OrbitControls';
+
+import { CameraManager, CameraManagerHelper, CameraState, CameraChangeDelegate } from '@cognite/reveal';
+
+export class CustomCameraManager implements CameraManager {
+    private _domElement: HTMLElement;
+    private _camera: THREE.PerspectiveCamera;
+    private _controls: OrbitControls;
+    private _cameraManagerHelper = new CameraManagerHelper();
```

Comment:
Can we not make this a static class so we instead do `CameraManagerHelper.helpMe()`?

Reply:
Yeah, will do that.